test/test_dnsserver.rb

```diff
@@ -91,11 +92,13 @@ def selector_thread
     @rcv_thread = Thread.new do
       loop do
         begin
+          # NOTE: ret value ignored below; did you mean to call 'each'?
           @selector.select { |monitor| monitor.value.call(monitor) }
           if @selector.closed?
```

Comment:
This is select is a blocking method that does something similar to io or kernel select.
It's different from enumerable select.